tests/foreman/api/test_syncplan.py

```diff
@@ -371,21 +371,21 @@ def test_positive_update_interval(self):
         :CaseImportance: Critical
         """
         for interval in valid_sync_interval():
-            with self.subTest(interval):
-                sync_plan = entities.SyncPlan(organization=self.org)
-                result = sync_plan.get_fields()['interval']
-                sync_plan.interval = sample(
-                    set(result.choices) - set([interval]),
-                    1
-                )[0]
-                sync_plan = sync_plan.create()
-                sync_plan.interval = interval
-                if (interval == 'custom cron'):
-                    sync_plan.cron_expression = gen_choice(valid_cron_expressions())
-                    sync_plan = sync_plan.update(['interval', 'cron_expression'])
-                else:
-                    sync_plan = sync_plan.update(['interval'])
-                self.assertEqual(sync_plan.interval, interval)
+            sync_plan = entities.SyncPlan(
+                description=gen_string('alpha'),
+                organization=self.org,
+                interval=interval
+            )
+            if interval == SYNC_INTERVAL['custom']:
+                sync_plan.cron_expression = gen_choice((valid_cron_expressions()))
```

Comment:
Do you need parentheses around `valid_cron_expressions()`? I didn't notice any difference between version with them and without them in Python shell.

Reply:
Re: parentheses around valid_cron_expressions()
I see its used in about half the occurrences of gen_choice. I will remove in my test. Would it be acceptable to remove them throughout the /tests/foreman/api/test_syncplan.py file? or just leave that for another issue?

Re: sync_plan.interval = something
The original idea was to use gen_choice as its already used in this test. The example you gave would not be random would it. But valid_intervals.remove(interval) would use a value not already in use, which is the important point here; its what we want. I will try that. 

Thank you

Reply:
My flake 8 test is passing, must be a config issue on my side.

Reply:
I agree, using `gen_choice` sounds like better idea than my proposal above.

Re: parentheses - I only saw your patch and noticed they weren't there originally, but are now. If there are other places in that file, or Robottelo, where they are around `valid_cron_expressions()`, then I think having them here is fine. They aren't really wrong - just unnecessary, as far as I can tell. That's why flake8 won't catch them.

If you want to remove them all from this file, personally I would do it in another PR. I think it's easier for reviewers to focus on one thing at a time. But nobody will complain if you do it in another commit in this PR.

Reply:
I would like to get your method working first, its invaluable experience from me. I just made that commit.

If we use gen_choice, we would still need to ensure we had a new interval.

Reply:
That's what `valid_intervals.remove(interval)` is for. After that line, we are sure that `valid_intervals` does not include interval used during sync_plan.create(). You are safe to use `gen_choice` on that list.

Reply:
I understand that line `valid_intervals.remove(interval)`, I was just wondering how many lines we save by using gen_choice  if it cannot do that as well. It seems to have other error handling benefits:
https://fauxfactory.readthedocs.io/en/latest/_modules/fauxfactory/factories/choices.html?highlight=gen_choice